### PR TITLE
Initialize prow for kapp controller openshift

### PR DESCRIPTION
**Pull request by theishshah**
Adding ci config files and openshift scaffolding for prow.

### CODE REVIEW

**Review by m1kola — COMMENTED** [review]
It looks like owner files are missing.

Note that prow by default expects `OWNERS` and `OWNERS_ALIASES`, but in other OLM downstream repos we call them`DOWNSTREAM_OWNERS` and `DOWNSTREAM_OWNERS_ALIASES` (see an example in https://github.com/openshift/operator-framework-operator-controller). This requires a bit of extra configuration in openshift/release.

Some refs:
* https://github.com/openshift/release/blob/a5767aa630146f7f1fff941f08f58649cdb60efa/core-services/prow/02_config/_plugins.yaml#L696C15-L707
* https://docs.ci.openshift.org/docs/how-tos/onboarding-a-new-component/
* https://docs.prow.k8s.io/docs/components/plugins/approve/approvers/

**Comment by joelanford** [comment]
Also, please amend the commit message to be `UPSTREAM: <carry>: add openshift-specific files` or something along those lines. The `UPSTREAM: <carry>:` will be important for future syncing automation.

File: openshift/go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module _
+
+go 1.20
+
+require github.com/openshift/build-machinery-go v0.0.0-20231128094528-1e9b1b0595c8 // indirect
```
[the remark this answers is not in the document]
Comment (theishshah) [reply openshift/go.mod:5]: This openshift folder was generation/fetched from github.com/openshift/build-machinery-go

Should the modification be made up there or just in our sample?

**Comment by theishshah** [comment]
> Also, please amend the commit message to be `UPSTREAM: <carry>: add openshift-specific files` or something along those lines. The `UPSTREAM: <carry>:` will be important for future syncing automation.

Will do, planning on squashing this whole thing into one for the merge and will name that accordingly

File: openshift/go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (m1kola) [reply openshift/go.mod:5]: Hm. If it is auto generated - then I don't how to answer this... It is just weird that `dependencies.go` is empty. For example, take a look at similar commit in the operator-controller repo: https://github.com/openshift/operator-framework-operator-controller/commit/2f59dfd234da3b172bff2a1bc1c0e98651d47b4c. It is not empty there.

File: openshift/go.mod [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module _
+
+go 1.20
```
Comment (grokspawn) [inline openshift/go.mod:3]: Shouldn't this be 1.21 to agree with [this](https://github.com/openshift/operator-framework-kapp-controller/pull/12/files#diff-4d5699538541ad6da5f6ae79150b49a432b610f4cba488cec683ac9dd00b18d2R4)?

File: openshift/go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (grokspawn) [reply openshift/go.mod:5]: IIRC, Andy added that to dependencies.go manually for catalogd.  I did the same for op-con.  You can copy.

**Comment by grokspawn** [comment]
Changes look good.  I think we're just needing [this](https://github.com/openshift/operator-framework-kapp-controller/pull/12#issuecomment-1984197793) to cross the finish line.

**Comment by theishshah** [comment]
All rolled up into one commit

**Comment by theishshah** [comment]
@joelanford if you get a chance to review so we can merge :)

**Comment by joelanford** [comment]
Nit: can you squash down to a single commit?

**Comment by grokspawn** [comment]
Unrelated to this, so if we want to do as a follow-up, cool... but we do not support github actions in downstream v1 components.  We should delete the .github directory.

**Comment by m1kola** [comment]
> Strike that. It appears that there is some automation which might handle that, once prow config is set up in openshift/release (see: [openshift/operator-framework-operator-controller@71ce53f](https://github.com/openshift/operator-framework-operator-controller/commit/71ce53fb845941dd0a3eb06de9152713a44b49c3)).

Yep, it is part of what bumper/syncer is doing. See [this](https://github.com/openshift/operator-framework-tooling/blob/fec7110a66de3a6cdd30025a11d33a8f43272427/pkg/v1/cmd.go#L518-L527).

We can remove it manually (in a separate PR) for now so we don't get annoyed by the dependabot PRs. And we will get the automation once we integrate with the syncer (separate ticket [here](https://issues.redhat.com/browse/OPRUN-3208)).

**Review by m1kola — APPROVED** [review]

**Pull request merged**